packages/hub/module-service.ts

```diff
@@ -20,10 +20,17 @@ export class ModuleService implements ModuleLoader {
   async load(card: Card, localModulePath: string, exportedName = 'default'): Promise<any> {
     // using md5 because this is just for cache validation, not cryptographic
     // collision resistance
-    let hash = createHash('md5');
-    hash.update(stringify(await toIdempotentDoc(card)));
-    let cardDir = join(cardFilesCache, hash.digest('hex'));
-    await this.cachedWriteCard(card, cardDir);
+
+    let cardDir;
+
+    if (card.cardDir) {
+      cardDir = card.cardDir;
```

Comment:
This is the only legitimate place where anybody should access `card.cardDir`. It would be nice to limit it more clearly in the type system. A challenge is that this state needs to persist in the database and persist across card cloning, so it's not sufficient to use something like a WeakMap on the Card instances.

Given that many developers are likely to encounter the `Card` class in the future and none of them should ever really see or think about `cardDir`, it's probably worth hiding it better. We could just do a type-unsafe cast here at the one spot where we legitimately access it:

```js
cardDir = (card as any)?.meta?.cardDir
```

The downside is that this won't generate a compile error if `Card` changes to not have a private `meta` anymore. But as long as we cover it with a test I'm fine with that.

Reply:
I filed a separate issue to track this since this PR is merged already https://github.com/cardstack/cardstack/issues/1404